app/Models/Race.php

```diff
@@ -207,6 +212,27 @@ public function filterableColumns(): array
     {
         return [
             'race_id',
+            'locations_id'
         ];
     }
+    /**
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsToMany
+     */
+    public function locations()
+    {
+        return $this->belongsToMany('App\Models\Location', 'race_location');
+    }
+    /**
+     * Determine if the model has profile data to be displayed
+     * @return bool
+     */
+    public function showProfileInfo(): bool
+    {
+
+        if ($this->locations) {
+            return true;
+        }
+
+        return false;
```

Comment:
The check on the type has gone. Might be simpler to return parent::showProfile() instead of return false, to use the basic showProfile function from the MiscModel